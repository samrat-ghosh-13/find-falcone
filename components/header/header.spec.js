// react
import React from "react";

// next
import router from "next/router";

// component
import Header from "./header";

// jest renderer
import renderer from "react-test-renderer";

// testing library react
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

jest.mock("next/router", () => require("next-router-mock"));

describe("Test Header component", () => {
  it("Case 1: Header Component Renders Correctly (Snapshot)", () => {
    const headerComponentTree = renderer.create(<Header />).toJSON();
    expect(headerComponentTree).toMatchSnapshot();
  });
  it("Case 2: Checks the header brand is correct or not", () => {
    const { getByText } = render(<Header />);
    expect(getByText("Finding Falcone", { exact: true })).toBeInTheDocument();
  });
  it("Case 3: Checks the planets button click redirects to the correct route or not", () => {
    const { getByText } = render(<Header />);
    fireEvent.click(getByText("Planets"));
    expect(router).toMatchObject({
      asPath: "/planets",
      pathname: "/planets",
      query: {},
    });
  });
  it("Case 4: Checks the vehicles button click redirects to the correct route or not", () => {
    const { getByText } = render(<Header />);
    fireEvent.click(getByText("Vehicles"));
    expect(router).toMatchObject({
      asPath: "/vehicles",
      pathname: "/vehicles",
      query: {},
    });
  });
  it("Case 5: Checks the Header Component has the planets and vehicles button", () => {
    const { getByText } = render(<Header />);
    expect(getByText("Planets", { exact: true })).toBeInTheDocument();
    expect(getByText("Vehicles", { exact: true })).toBeInTheDocument();
  });
});
