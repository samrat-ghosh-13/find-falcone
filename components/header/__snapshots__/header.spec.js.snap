// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component Case 1: Header Component Renders Correctly (Snapshot) 1`] = `
<article>
  <header
    className="app__header"
  >
    <div
      className="app__header__contents"
    >
      <div
        className="app__header__contents__left"
        onClick={[Function]}
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": 32,
              "overflow": "hidden",
              "position": "relative",
              "width": 32,
            }
          }
        >
          <noscript>
            <img
              alt="Falcons Icon"
              decoding="async"
              src="/_next/image?url=%2Fassets%2Ffalcons.png&w=64&q=75"
              srcSet="/_next/image?url=%2Fassets%2Ffalcons.png&w=32&q=75 1x, /_next/image?url=%2Fassets%2Ffalcons.png&w=64&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
          <img
            alt="Falcons Icon"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </div>
        <p
          className="app__header__contents__left__text"
        >
          Finding Falcone
        </p>
      </div>
      <div
        className="app__header__contents__right"
      >
        <button
          buttons={Array []}
          className=" button app__header__contents__right__cta undefined  undefined"
          data-testid="button"
          onClick={[Function]}
        >
          <span
            className="button__text"
            data-testid="button__text"
          >
            Planets
          </span>
        </button>
        <button
          buttons={Array []}
          className=" button app__header__contents__right__cta undefined  undefined"
          data-testid="button"
          onClick={[Function]}
        >
          <span
            className="button__text"
            data-testid="button__text"
          >
            Vehicles
          </span>
        </button>
      </div>
    </div>
  </header>
</article>
`;
